src/code/z_lifemeter.c

```diff
@@ -1,17 +1,446 @@
+    interfaceCtx->lifeColorChange = 0;
+    interfaceCtx->lifeColorChangeDirection = 0;
+
+    interfaceCtx->lifeSizeChange = interfaceCtx->lifeColorChange;
+    interfaceCtx->lifeSizeChangeDirection = interfaceCtx->lifeColorChangeDirection;
+
+    interfaceCtx->heartsPrimR[0] = 0xFF;
+    interfaceCtx->heartsPrimG[0] = 0x46;
+    interfaceCtx->heartsPrimB[0] = 0x32;
+
+    interfaceCtx->heartsEnvR[0] = 0x32;
+    interfaceCtx->heartsEnvG[0] = 0x28;
+    interfaceCtx->heartsEnvB[0] = 0x3C;
+
+    interfaceCtx->heartsPrimR[1] = 0xFF;
+    interfaceCtx->heartsPrimG[1] = 0x46;
+    interfaceCtx->heartsPrimB[1] = 0x32;
+
+    interfaceCtx->heartsEnvR[1] = 0x32;
+    interfaceCtx->heartsEnvG[1] = 0x28;
+    interfaceCtx->heartsEnvB[1] = 0x3C;
+
+    sHeartsDDPrim[0][0] = sHeartsDDPrim[1][0] = 0xFF;
+    sHeartsDDPrim[0][1] = sHeartsDDPrim[1][1] = 0xFF;
+    sHeartsDDPrim[0][2] = sHeartsDDPrim[1][2] = 0xFF;
+
+    sHeartsDDEnv[0][0] = sHeartsDDEnv[1][0] = 0xC8;
+    sHeartsDDEnv[0][1] = sHeartsDDEnv[1][1] = 0;
+    sHeartsDDEnv[0][2] = sHeartsDDEnv[1][2] = 0;
+}
+
+void LifeMeter_UpdateColors(GlobalContext* globalCtx) {
+    InterfaceContext* interfaceCtx = &globalCtx->interfaceCtx;
+    f32 factorBeating = interfaceCtx->lifeColorChange * 0.1f;
+    f32 ddFactor;
+    s32 type = 0;
+    s32 ddType;
+    s16 rFactor;
+    s16 gFactor;
+    s16 bFactor;
+
+    if (interfaceCtx) {}
+
+    if (interfaceCtx->lifeColorChangeDirection != 0) {
+        interfaceCtx->lifeColorChange--;
+        if (interfaceCtx->lifeColorChange <= 0) {
+            interfaceCtx->lifeColorChange = 0;
+            interfaceCtx->lifeColorChangeDirection = 0;
+        }
+    } else {
+        interfaceCtx->lifeColorChange++;
+        if (interfaceCtx->lifeColorChange >= 10) {
+            interfaceCtx->lifeColorChange = 10;
+            interfaceCtx->lifeColorChangeDirection = 1;
+        }
+    }
+
+    ddFactor = factorBeating;
+
+    interfaceCtx->heartsPrimR[0] = 0xFF;
+    interfaceCtx->heartsPrimG[0] = 0x46;
+    interfaceCtx->heartsPrimB[0] = 0x32;
+
+    interfaceCtx->heartsEnvR[0] = 0x32;
+    interfaceCtx->heartsEnvG[0] = 0x28;
+    interfaceCtx->heartsEnvB[0] = 0x3C;
+
+    interfaceCtx->heartsPrimR[1] = sHeartsPrimColors[type][0];
+    interfaceCtx->heartsPrimG[1] = sHeartsPrimColors[type][1];
+    interfaceCtx->heartsPrimB[1] = sHeartsPrimColors[type][2];
+
+    interfaceCtx->heartsEnvR[1] = sHeartsEnvColors[type][0];
+    interfaceCtx->heartsEnvG[1] = sHeartsEnvColors[type][1];
+    interfaceCtx->heartsEnvB[1] = sHeartsEnvColors[type][2];
+
+    rFactor = sHeartsPrimFactors[0][0] * factorBeating;
+    gFactor = sHeartsPrimFactors[0][1] * factorBeating;
+    bFactor = sHeartsPrimFactors[0][2] * factorBeating;
+
+    interfaceCtx->beatingHeartPrim[0] = (u8)(rFactor + 0xFF) & 0xFF;
+    interfaceCtx->beatingHeartPrim[1] = (u8)(gFactor + 0x46) & 0xFF;
+    interfaceCtx->beatingHeartPrim[2] = (u8)(bFactor + 0x32) & 0xFF;
```

Comment:
```suggestion
    interfaceCtx->beatingHeartPrim[0] = (u8)(rFactor + 255);
    interfaceCtx->beatingHeartPrim[1] = (u8)(gFactor + 70);
    interfaceCtx->beatingHeartPrim[2] = (u8)(bFactor + 50);
```
Are the `& 0xFF` needed to match? The `(u8)` should take care of that but also IDO can be a bit silly sometimes. This comment applies to a few places

Reply:
yes, the  `& 0xFF` are needed to match